numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2406,6 +2422,46 @@ def get_function(self, name):
     def get_global_symbol(self, name):
         """Return a MemoryPointer referring to the named symbol"""
 
+    def _set_callables(self, callbacks, attr):
+        if not isinstance(callbacks, list):
+            raise TypeError("callbacks must be a list")
+
+        for f in callbacks:
+            if not callable(f):
+                raise TypeError("callback must be callable")
+
+        setattr(self, attr, callbacks)
+
+    def setup(self):
+        """Call the setup functions for cumodule with the given `stream`"""
+        if self.setup_functions is None or self.initialized:
```

Comment:
Should it be an error to call `setup()` twice? The reason I ask is that there's an asymmetry between `setup()` and `_set_finalizers()` - Calling `setup()` twice will result in the setup functions being called once. But calling `_set_finalizers()` twice looks like it will result in the teardown callbacks being run twice at module unload time.

Reply:
I gues `_set_finalizers()` isn't invoked from outside the class, so maybe for that reason the asymmetry doesn't matter.

Reply:
I prefer raising an error. The current state quietly skips setup function if it's been initialized once, which may go against what user would expect.